### PR TITLE
bug fix (omp with large #threads)

### PR DESCRIPTION
**Pull request by liruipeng**
Fixed a problem in `hypre_ParCSRMatrixGenerateFFFC` with OMP, when `OMP_NUM_THREADS` is sufficiently large. This was accidentally found when doing tests on Lassen with `srun`, where `OMP_NUM_THREADS` was automatically set to be `160`!

### CODE REVIEW

File: src/parcsr_mv/gen_fffc.c [hunk 1]
```diff
@@ -346,6 +346,12 @@ hypre_ParCSRMatrixGenerateFFFC( hypre_ParCSRMatrix  *A,
          {
             fpt = fpt_array[i];
             fpt2 = fpt_array[i-1];
+
```
Comment (liruipeng) [inline src/parcsr_mv/gen_fffc.c:349]: A simple fix but hard to find. In the cases that `num_threads` is large enough, a thread can have no F-pt in its "partition", so we must skip the adds in the rest of the loop.

File: src/parcsr_mv/gen_fffc.c [hunk 2]
```diff
@@ -346,6 +346,12 @@ hypre_ParCSRMatrixGenerateFFFC( hypre_ParCSRMatrix  *A,
...
+            if (fpt == fpt2)
```
Comment (liruipeng) [inline src/parcsr_mv/gen_fffc.c:350]: A simple fix but hard to find. In the cases that num_threads is large enough, a thread can have no F-pt in its "partition", so we must skip the adds in the rest of the loop.

**Review by rfalgout — APPROVED** [review]
Looks good to me.

**Review by ulrikeyang — APPROVED** [review]
We probably should check if we have the same problem somewhere else.

**Pull request merged**